File: examples/CircadianVariability/CircadianVariability.m
% This file is part of LoopInsighT1, an open source tool to
% simulate closed-loop glycemic control in type 1 diabetes.
% Distributed under the MIT software license.
% See https://lt1.org for further information.


opts = delimitedTextImportOptions("NumVariables", 11);
opts.DataLines = [2, Inf];
opts.Delimiter = ",";
opts.VariableTypes = ["double", "double", "double", "double", "double", "double", "double", "double", "double", "double", "double"];

data_table = readtable("CircadianVariability.csv", opts);

data = table2array(data_table);

plot(data(:,1) / 60, data(:,2:end))

File: examples/CircadianVariability/CircadianVariability.py
# This file is part of LoopInsighT1, an open source tool to
# simulate closed-loop glycemic control in type 1 diabetes.
# Distributed under the MIT software license.
# See https://lt1.org for further information.


import matplotlib.pyplot as plt
import numpy as np
import csv

with open('CircadianVariability.csv', 'r', newline='') as csvfile:
    reader = csv.reader(csvfile)
    data = list(reader)
    csvfile.close()

    
data_array = np.array(data[1:], dtype=float)
for i in range(1,len(data[0])):
    print(i)
    plt.plot(data_array[:,0]/60, data_array[:,i])
        
plt.show()
    
